src/nft-card.ts

```diff
@@ -190,24 +189,37 @@ export class NftCard extends LitElement {
               .traitData=${this.traitData}
             ></nft-card-back>
             `
-          }
+        }
        </div>
      `
-   }
-
-  /**
-	  * async connectWallet - Initializes connection to the injected web3 account
-	  */
-  private async connectWallet() {
-    if (!window.web3) {
-      //
-    } else if (window.ethereum) {
-      window.ethereum.enable()
-    } else {
-      const errorMessage =
-        'You need an Ethereum wallet to interact with this marketplace. Unlock your wallet, get MetaMask.io or Portis on desktop, or get Trust Wallet or Coinbase Wallet on mobile.'
-      alert(errorMessage)
-      throw new Error(errorMessage)
     }
-  }
+
+    private async getAsset(contractAddress, tokenId) {
+        this.asset = await this.seaport.api.getAsset(contractAddress, tokenId);
+    }
+
+    private getNetwork() {
+        switch (this.network) {
+            case 'main':
+                return Network.Main
+            case 'rinkeby':
+                return Network.Rinkeby
+        }
+    }
+
+    /**
+     * async connectWallet - Initializes connection to the injected web3 account
+     */
+    private async connectWallet() {
+        if (!window.web3) {
+            //
+        } else if (window.ethereum) {
+            window.ethereum.enable()
+        } else {
```

Comment:
we were gonna do this check at a higher level and show a `View on OpenSea` if there's no wallet, right?

Reply:
Yup!